### PR TITLE
lib/raster: Fix Resource Leak issue in put_title.c

### PR DESCRIPTION
**Pull request by ShubhamDesai**
This pull request fixes issue identified by Coverity Scan (CID: 1207660)
used remove() to fix this issue.

### CODE REVIEW

**Review by nilason — CHANGES_REQUESTED** [review]
Please see [G_tempfile()](https://grass.osgeo.org/programming8/defs_2gis_8h.html#a68db8506eea42bbffef9f5e58ef9fb73) for handling resources.

**Comment by ShubhamDesai** [comment]
> Please see [G_tempfile()](https://grass.osgeo.org/programming8/defs_2gis_8h.html#a68db8506eea42bbffef9f5e58ef9fb73) for handling resources.

yes it says free() can be used so in this case G_free() should work fine after remove()

File: lib/raster/put_title.c [hunk 1]
```diff
@@ -35,6 +35,8 @@ int Rast_put_cell_title(const char *name, const char *title)
     if (!out) {
         fclose(in);
         G_warning(_("G_put_title - can't create a temp file"));
+        remove(tempfile);
```
[the remark this answers is not in the document]
Comment (ShubhamDesai) [reply lib/raster/put_title.c]: done

[the remark this answers is not in the document]
Comment (nilason) [reply lib/raster/put_title.c]: Now I see there are two more if statements that return, which both need some care.

[the remark this answers is not in the document]
Comment (ShubhamDesai) [reply lib/raster/put_title.c]: added G_free(tempfile) in both ifs

[the remark this answers is not in the document]
Comment (nilason) [reply lib/raster/put_title.c]: Great!

File: lib/raster/put_title.c [hunk 2]
```diff
@@ -66,6 +70,7 @@ int Rast_put_cell_title(const char *name, const char *title)
         fclose(in);
         G_warning(_("can't write category information for [%s] in [%s]"), name,
                   mapset);
+        G_free(tempfile);
```
Comment (nilason) [inline lib/raster/put_title.c:73]: Here tempfile may be removed, (`in`) is successfully assigned.

Comment (ShubhamDesai) [reply lib/raster/put_title.c:74]: done

**Review by nilason — APPROVED** [review]

**Pull request merged**